### PR TITLE
Add benchmark organization guide

### LINKED ISSUE #32: Add user guide: Best practices on modularizing benchmark code
Add user guide to the documentation:
- [ ] Best practices on modularizing benchmark code (split into different files, benchmark organization based on semantic groups, reusing setup/teardown code, ...)

### PR DESCRIPTION
**Pull request by nicholasjng**
Contains the two most important structuring tips - refactoring into a proper directory and semantic grouping - in a user guide document.

Also introduces tags as another layer of organization that allows grouping across different module files.

I based this on the MNIST example #11 to take over those changes for the docs organization, so that needs to go in before this.

Closes #32.

### CODE REVIEW

File: docs/guides/organization.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# How to efficiently organize benchmark code
+
+To efficiently organize benchmarks and keeping your setup modular, you can follow a few guidelines.
```
Comment (maxmynter) [inline docs/guides/organization.md:3]:
```suggestion
To efficiently organize benchmarks and keeping your setup modular, we propose the following guidelines.
```

File: docs/guides/organization.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+## Tip 3: Attach tags to benchmarks for selective filtering
+
+For structuring benchmarks within files, you can also use **tags**, which are tuples of strings attached to a benchmark:
+
+```python
+# benchmarks/data_quality.py
+import nnbench
+
+
+@nnbench.benchmark(tags=("foo",))
```
Comment (maxmynter) [inline docs/guides/organization.md:58]: Should we add a small note like:

> Note the trailing comma. This is necessary for Python to correctly parse single entry tuples as such. Without it, the value will be interpreted as a string, causing a type error.

**Review by maxmynter — COMMENTED** [review]
Nice. Short, to the point but valuable.

File: docs/guides/organization.md (hunk 2 above)
Comment (nicholasjng) [reply docs/guides/organization.md:58]: That's not really our issue, IDEs complain when you leave out the comma as far as I know.

**Pull request merged**